algorithmic_efficiency/workloads/librispeech_conformer/librispeech_jax/models.py

```diff
@@ -0,0 +1,615 @@
+  computing mean and variance.
+
+  This is inspired by lingvo jax implementation of BatchNorm:
+  https://github.com/tensorflow/lingvo/blob/84b85514d7ad3652bc9720cb45acfab08604519b/lingvo/jax/layers/normalizations.py#L92
+
+  and the corresponding defaults for momentum and epsilon have been copied over
+  from lingvo.
+  """
+  config: ConformerConfig
+
+  def setup(self):
+    dim = self.config.encoder_dim
+    dtype = self.config.dtype
+
+    self.ra_mean = self.variable('batch_stats', 'mean',
+                                 lambda s: jnp.zeros(s, dtype), dim)
+    self.ra_var = self.variable('batch_stats', 'var',
+                                lambda s: jnp.ones(s, dtype), dim)
+
+    self.gamma = self.param('scale', nn.initializers.zeros, dim, dtype)
+    self.beta = self.param('bias', nn.initializers.zeros, dim, dtype)
+
+  @nn.compact
+  def __call__(self, inputs, input_paddings, train):
+    rank = inputs.ndim
+    reduce_over_dims = list(range(0, rank - 1))
+
+    padding = jnp.expand_dims(input_paddings, -1)
+    momentum = self.config.batch_norm_momentum
+    epsilon = self.config.batch_norm_epsilon
+
+    if train:
+      mask = 1.0 - padding
+      sum_v = jnp.sum(inputs * mask, axis=reduce_over_dims, keepdims=True)
+      count_v = jnp.sum(
+          jnp.ones_like(inputs) * mask, axis=reduce_over_dims, keepdims=True)
+
+      count_v = jnp.maximum(count_v, 1.0)
+      mean = sum_v / count_v
+
+      sum_vv = jnp.sum(
+          (inputs - mean) * (inputs - mean) * mask,
+          axis=reduce_over_dims,
+          keepdims=True)
+
+      var = sum_vv / count_v
+
+      self.ra_mean.value = momentum * self.ra_mean.value + (
+          1 - momentum) * mean
+      self.ra_var.value = momentum * self.ra_var.value + (
+          1 - momentum) * var
+    else:
+      mean = self.ra_mean.value
+      var = self.ra_var.value
+
+    inv = (1 + self.gamma) / jnp.sqrt(var + epsilon)
+
+    bn_output = (inputs - mean) * inv + self.beta
+    bn_output *= 1.0 - padding
+
+    return bn_output
+
+
+class ConvolutionBlock(nn.Module):
+  r"""Convolution block in conformer layer.
+
+  architecture:
+
+   input                   # (batch, time, hidden_dim)
+     |
+   layer_norm(.)           # (batch, time, hidden_dim)
+   dense(.), dense(.)      # (batch, time, 2 * hidden_dim)
+     |      /
+    glu(.)                 # (batch, time, hidden_dim)
+    depthwise_conv1d(.)
+    batch_norm(.)
+    act(.)
+     |
+    dense(.)
+    dropout(.)
+     |
+    output
+  """
+  config: ConformerConfig
+
+  @nn.compact
+  def __call__(self, inputs, input_paddings, train):
+    config = self.config
+    inputs = LayerNorm(dim=config.encoder_dim)(inputs)
+
+    input_gated1 = nn.Dense(
+        config.encoder_dim,
+        kernel_init=nn.initializers.xavier_uniform(),
+        use_bias=True)(inputs)
+
+    input_gated2 = nn.Dense(
+        config.encoder_dim,
+        kernel_init=nn.initializers.xavier_uniform(),
+        use_bias=True)(inputs)
```

Comment:
I haven't tried this , will try it and update if it's faster in future PR 
I did switch encoder_dim and batch size (512 dim 256 batch size instead of the other way around) and that led to 99% GPU utilization as compared to ~50% in default configuration